Makefile

```diff
@@ -133,6 +160,24 @@ iso: check $(PHOTON_STAGE) $(PHOTON_PACKAGES) ostree-repo
                 -f > \
                 $(PHOTON_LOGS_DIR)/installer.log 2>&1
 
+full-dep-list: $(PHOTON_MINIMAL_DEPLIST) 
+	@echo "Generating the install time dependency list for full installation"
+	@cd $(PHOTON_SPECDEPS_DIR) && \
+		$(PHOTON_SPECDEPS) -s $(PHOTON_SPECS_DIR) --input-type=json --file packages_full.json -d json -a $(PHOTON_DATA_DIR)
+	@echo "generated the json file with all dependencies for full installation"
+
+deptree-full:
+	@cd $(PHOTON_SPECDEPS_DIR) && \
+		$(PHOTON_SPECDEPS) -s $(PHOTON_SPECS_DIR) -i json -f packages_full.json -a $(PHOTON_DATA_DIR)
+
+deptree:
+	@cd $(PHOTON_SPECDEPS_DIR) && \
+		$(PHOTON_SPECDEPS) -s $(PHOTON_SPECS_DIR) -i pkg -p $(pkg)
+
+who-needs:
```

Comment:
These are meant for our understanding of package dependencies. They are independent targets that displays the dependencies.